### PR TITLE
Clean lower level motor control

### PR DESCRIPTION
**Pull request by muscatd75**

### CODE REVIEW

File: codeGenerator/src/CoreCodeGenerator/baseGenerator.cs [hunk 1]
```diff
@@ -140,6 +140,10 @@ internal void copyrightAndGenNoticeAndSave(string outputFilePathName, string con
 
                     if (currentText == contentsWithoutGenInfo)
                         writeFile = false;
+                    else
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply codeGenerator/src/CoreCodeGenerator/baseGenerator.cs:143]: done

File: codeGenerator/templates/MechanismInstance.h [hunk 2]
```diff
@@ -27,15 +27,9 @@ class $$_MECHANISM_INSTANCE_NAME_$$ : public $$_MECHANISM_INSTANCE_NAME_$$Gen
...
+    void RunCommonTasks() override;
 
-    // todo not sure what to do with these
-    /*
-    bool IsAtMinPosition(RobotElementNames::ROBOT_ELEMENT_NAMES identifier) const override;
-    bool IsAtMinPosition(RobotElementNames::ROBOT_ELEMENT_NAMES identifier) const override;
-    bool IsAtMaxPosition(RobotElementNames::ROBOT_ELEMENT_NAMES identifier) const override;
-    bool IsAtMaxPosition(RobotElementNames::ROBOT_ELEMENT_NAMES identifier) const override;
-    */
+    void createAndRegisterStates();
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply codeGenerator/templates/MechanismInstance.h:32]: done

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp [hunk 3]
```diff
@@ -57,31 +58,35 @@ ClimberManager::ClimberManager ( ClimberManagerGen *base ) : ClimberManagerGen()
...
+void ClimberManager::RunCommonTasks()
 {
-	OffState* OffStateInst = new OffState ( string ( "Off" ), 0, new ClimberManagerAllStatesStateGen ( string ( "Off" ), 0, this ), this );
-	AddToStateVector ( OffStateInst );
+	// This function is called once per loop before the current state Run()
+}
 
-	InitializeState* InitializeStateInst = new InitializeState ( string ( "Initialize" ), 1, new ClimberManagerAllStatesStateGen ( string ( "Initialize" ), 1, this ), this );
-	AddToStateVector ( InitializeStateInst );
+void ClimberManager::createAndRegisterStates()
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp:66]: done

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp [hunk 4]
```diff
@@ -44,10 +44,11 @@ using namespace ClimberManagerStates;
 /// @param otherMotor Same as previous
 /// @param solenoid Solenoid in the mechanism - code generator should probably use the usage for the variable name
 /// Additional actuators and sensors are also in this list.
-ClimberManager::ClimberManager ( ClimberManagerGen *base ) : ClimberManagerGen(),IRobotStateChangeSubscriber(),
-	m_ClimberManager ( base )
+ClimberManager::ClimberManager(ClimberManagerGen *base) : ClimberManagerGen(), IRobotStateChangeSubscriber(),
+														  m_ClimberManager(base)
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp:48]: ClimberManager comes directly from what we entered in the GUI. I wrote an off season task to do formatting in the code generator

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp [hunk 5]
```diff
@@ -57,31 +58,35 @@ ClimberManager::ClimberManager ( ClimberManagerGen *base ) : ClimberManagerGen()
...
-	InitializeState* InitializeStateInst = new InitializeState ( string ( "Initialize" ), 1, new ClimberManagerAllStatesStateGen ( string ( "Initialize" ), 1, this ), this );
-	AddToStateVector ( InitializeStateInst );
+void ClimberManager::createAndRegisterStates()
+{
+	OffState *OffStateInst = new OffState(string("Off"), 0, new ClimberManagerAllStatesStateGen(string("Off"), 0, this), this);
+	AddToStateVector(OffStateInst);
 
-	ManualState* ManualStateInst = new ManualState ( string ( "Manual" ), 2, new ClimberManagerAllStatesStateGen ( string ( "Manual" ), 2, this ), this );
-	AddToStateVector ( ManualStateInst );
+	InitializeState *InitializeStateInst = new InitializeState(string("Initialize"), 1, new ClimberManagerAllStatesStateGen(string("Initialize"), 1, this), this);
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.cpp:71]: Yes it is dependent on how Tanay typed them. I wrote a story for the code gen to be updated in off season

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.h [hunk 6]
```diff
@@ -40,18 +40,19 @@ class ClimberManager : public ClimberManagerGen,public IRobotStateChangeSubscrib
...
 	ClimberManager() = delete;
 	~ClimberManager() = default;
 
+	void RunCommonTasks() override;
+
 	void createAndRegisterStates();
 
 	void Update(RobotStateChanges::StateChange change, int value) override;
 
 	bool isClimbMode() const { return m_climbMode == RobotStateChanges::ClimbMode::ClimbModeOn; }
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.h:53]: createAndRegisterStates is corrected in the templates. Will be fixed the next time that we regenerate

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.h [hunk 7]
```diff
@@ -40,18 +40,19 @@ class ClimberManager : public ClimberManagerGen,public IRobotStateChangeSubscrib
...
 	/// @param solenoid Solenoid in the mechanism - code generator should probably use the usage for the variable name
 	/// Additional actuators and sensors are also in this list.
-	ClimberManager ( ClimberManagerGen *generatedMech );
+	ClimberManager(ClimberManagerGen *generatedMech);
 	ClimberManager() = delete;
 	~ClimberManager() = default;
 
+	void RunCommonTasks() override;
+
 	void createAndRegisterStates();
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/ClimberManager.h:49]: ok

File: src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp [hunk 8]
```diff
@@ -48,48 +48,53 @@ using namespace Thing1MechStates;
 /// @param otherMotor Same as previous
 /// @param solenoid Solenoid in the mechanism - code generator should probably use the usage for the variable name
 /// Additional actuators and sensors are also in this list.
-Thing1Mech::Thing1Mech ( Thing1MechGen *base ) : Thing1MechGen(),IRobotStateChangeSubscriber(),
-	m_Thing1Mech ( base )
+Thing1Mech::Thing1Mech(Thing1MechGen *base) : Thing1MechGen(), IRobotStateChangeSubscriber(),
+											  m_Thing1Mech(base)
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp:52]: will be fixed in off season

File: src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp [hunk 9]
```diff
@@ -48,48 +48,53 @@ using namespace Thing1MechStates;
...
+void Thing1Mech::RunCommonTasks()
 {
-	leftFrontCWState* leftFrontCWStateInst = new leftFrontCWState ( string ( "leftFrontCW" ), 0, new Thing1MechAllStatesStateGen ( string ( "leftFrontCW" ), 0, this ), this );
-	AddToStateVector ( leftFrontCWStateInst );
+	// This function is called once per loop before the current state Run()
+}
 
-	rightFrontCWState* rightFrontCWStateInst = new rightFrontCWState ( string ( "rightFrontCW" ), 1, new Thing1MechAllStatesStateGen ( string ( "rightFrontCW" ), 1, this ), this );
-	AddToStateVector ( rightFrontCWStateInst );
+void Thing1Mech::createAndRegisterStates()
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp:65]: ok

File: src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp [hunk 10]
```diff
@@ -48,48 +48,53 @@ using namespace Thing1MechStates;
...
+{
+	leftFrontCWState *leftFrontCWStateInst = new leftFrontCWState(string("leftFrontCW"), 0, new Thing1MechAllStatesStateGen(string("leftFrontCW"), 0, this), this);
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/Thing1Mech/decoratormods/Thing1Mech.cpp:67]: off season update to code gen

**Review by witcpalek — APPROVED** [review]

**Pull request merged**